codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/EndpointResolverMiddleware.kt

```diff
@@ -53,9 +53,12 @@ class EndpointResolverMiddleware(
         writer.write("let endpoint = try endpointResolver.resolve(params: endpointParams)")
             .write("")
 
-        writer.write("""let authScheme = endpoint.authScheme(name: "sigv4")""")
-
-        writer.write("""let awsEndpoint = AWSEndpoint(endpoint: endpoint, signingName: authScheme?["signingName"] as? String, signingRegion: authScheme?["signingRegion"] as? String)""")
+        writer.write("let authScheme = endpoint.authSchemes()?.first")
+        writer.write("let signingName = Endpoint.signingName(from: authScheme)")
+        writer.write("let signingRegion = Endpoint.signingRegion(from: authScheme)")
+        writer.write("let signingAlgorithm = Endpoint.signingAlgorithm(from: authScheme)")
+            .write("")
+        writer.write("let awsEndpoint = AWSEndpoint(endpoint: endpoint, signingName: signingName, signingRegion: signingRegion)")
             .write("")
 
         writer.write("""var host = """"")
```

Comment:
I know what this is doing, but that doesn't mean it doesn't make my head spin

Reply:
this is definitely better than escaping every quote. Specially good at you can just copy paste when you want to emit exact Swift code without any substitution.

Reply:
In this specific case I would consider it fine, maybe even better, to escape two double quote characters.  But no doubt that if I had a big blob of JSON or the like, containing numerous double quote characters, I would contain my string in triple quotes.  And don't get me started on including a literal `$` in a Kotlin multiline string...